### ISSUE TITLE
Cognito user pool domain incorrectly rejects fully-qualified domains

### ISSUE THREAD

**Issue by ThePletch** [terraform-linters/tflint-ruleset-aws#156]
The linting rules for the `aws_cognito_user_pool_domain` resource incorrectly restrict the `domain` attribute to consist only of the subdomain, but when specifying a fully-customized domain for a pool, you must specify the entire domain, including the root and its TLD.

Error example:
```hcl
resource "aws_cognito_user_pool_domain" "zombo" {
  domain          = "zombocom.zombo.com"
  certificate_arn = "whatever"
  user_pool_id    = aws_cognito_user_pool.zombocom.id
}
```
Produces:
```
Error: "zombocom.zombo.com" does not match valid pattern ^[a-z0-9](?:[a-z0-9\-]{0,61}[a-z0-9])?$ (aws_cognito_user_pool_domain_invalid_domain)
```

**Comment by ThePletch** [comment]
Example from the AWS Terraform provider docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cognito_user_pool_domain#custom-cognito-domain

**Comment by wata727** [comment]
Well, this rule is generated from the AWS API Reference:
https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPoolDomain.html#API_CreateUserPoolDomain_RequestParameters

I haven't confirmed this, but if this document is incorrect, we may need to remove this regexp-based rule.

**Comment by wata727** [comment]
See also https://github.com/terraform-linters/tflint/issues/933

This regex may be ignored internally only when using a fully-customized domain.
